lib/tasks_private/instance_types.rake

```diff
@@ -1,215 +1,191 @@
-    next unless available_instance
-    output_instances << available_instance
-    current_available.delete(type)
-    available.delete(type)
-  end
+    # using python sdk, as it contains more historical data than ruby-sdk
+    SDK_REPO_PATH = 'boto/botocore'     # 'aws/aws-sdk-ruby'
+    SDK_DATA_DIR  = 'botocore/data/ec2' # 'apis/ec2'
+    SDK_FILE_NAME = 'service-2.json'    # 'api-2.json'
 
-  # new available types
-  new_count = available.count
-  available.values.each do |instance|
-    output_instances << instance
-  end
+    GH_API_HOST = 'api.github.com'
+    GH_API_HEADERS = {
+      'Accept'        => 'application/vnd.github.v3+json',
+      'Authorization' => "token #{GH_API_TOKEN}",
+    }.freeze
 
-  new_instance_types_file << output_instances.map { |i| format_instance(i) }.join("\n")
-  new_instance_types_file << <<EOT
-  }.freeze
-
-  # Types that are still advertised, but not recommended for new instances.
-  DEPRECATED_TYPES = {
-EOT
-
-  # current deprecated types
-  output_instances = []
-  current_deprecated = ManageIQ::Providers::Amazon::InstanceTypes::DEPRECATED_TYPES.dup
-  current_deprecated.each do |type, instance|
-    next unless previous.key?(type)
-    output_instances << instance
-    current_deprecated.delete(type)
-    previous.delete(type)
-  end
+    GH_RAW_HOST = 'raw.githubusercontent.com'
 
-  # new deprecated types
-  deprecated_count = current_available.count
-  current_available.each do |type, instance|
-    next unless previous.key?(type)
-    output_instances << instance
-    current_available.delete(type)
-    previous.delete(type)
-  end
+    GH_CACHE = ActiveSupport::Cache::FileStore.new(Rails.root.join('tmp', 'aws_cache', 'github'))
+    GH_CACHE.logger = LOGGER
 
-  raise "#{previous.keys} are deprecated but have not been current before!" if previous.any?
-
-  new_instance_types_file << output_instances.map { |i| format_instance(i) }.join("\n")
-  new_instance_types_file << <<EOT
-  }.freeze
-
-  # Types that are no longer advertised
-  DISCONTINUED_TYPES = {
-EOT
-  # current discontinued types
-  output_instances = []
-  current_discontinued = ManageIQ::Providers::Amazon::InstanceTypes::DISCONTINUED_TYPES.dup
-  current_discontinued.values.each do |instance|
-    instance[:disabled] = true
-    output_instances << instance
-  end
+    JSON_CONTENT_RE = /\bapplication\/json\b/i
 
-  # new discontinued from current_available
-  discontinued_count = current_available.count
-  current_available.values.each do |instance|
-    instance[:disabled] = true
-    output_instances << instance
-  end
+    def http_get(uri, headers = {})
+      LOGGER.info("Making request to #{uri}")
 
-  # new discontinued from current_deprecated
-  discontinued_count += current_deprecated.count
-  current_deprecated.values.each do |instance|
-    instance[:disabled] = true
-    output_instances << instance
-  end
+      response = Net::HTTP.start(uri.host, :use_ssl => true) do |http|
```

Comment:
I thought elsewhere I mentioned using Octokit as opposed to Net::Http (unless you are grabbing non github content?

Reply:
I don't want to introduce a new gem for a simple `get` purpose

Reply:
@AlexanderZagaynov If you're just doing a simple get, then I recommend using `open-uri`.  It's part of stdlib, and greatly simplifies the code.  FWIW, I don't understand the aversion to adding a new gem, when the purpose of the gem makes the code clearer, even if it achieves ultimately the same thing.

Reply:
Since GitHub now isn't involved at all, we don't need that gem.
I believe `open-uri` can be a bad idea: https://sakurity.com/blog/2015/02/28/openuri.html
And I'm always avoiding such things just as pattern / good habit.